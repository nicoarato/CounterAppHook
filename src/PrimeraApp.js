import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitulo }) => {

    return (
        <>
            <h1>{saludo}!!</h1>
            {/* <p>{saludo}. {JSON.stringify(saludoJson)}</p> */}
            <p>{subtitulo}</p>
        </>
    )

};

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    saludo: 'Hola Mundo',
    subtitulo: 'soy un subtitulo'
}


export default PrimeraApp;