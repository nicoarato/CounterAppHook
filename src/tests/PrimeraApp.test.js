import React from 'react';

import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';

describe('Pruebas en <PrimeraApp>', () => {

    // test('Debe mostrar el Mensaje "Hola, Soy Goku"', () => {
    //     const saludo = 'Hola, soy Goku';
    //     const { getByText } = render(< PrimeraApp saludo={saludo} />);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })

    test('Debe mostrar <PrimeraApp /> correctamente', () => {

        const saludo = 'Hola, soy Goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />)

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar el subtitulo enviado por props.', () => {
        const saludo = 'Hola, soy Goku';
        const subtitulo = 'Hola, soy un subtitulo';
        const wrapper = shallow(
            <PrimeraApp
                saludo={saludo}
                subtitulo={subtitulo}
            />
        );

        const textoParrafo = wrapper.find('p').text();
        // console.log(textoParrafo);


        expect(textoParrafo).toBe(subtitulo);
    });




})