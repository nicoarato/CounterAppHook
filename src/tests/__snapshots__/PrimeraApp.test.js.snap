// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp> Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
    !!
  </h1>
  <p>
    soy un subtitulo
  </p>
</Fragment>
`;

exports[`Pruebas en <PrimeraApp> Debe mostrar el subtitulo enviado por props. 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
    !!
  </h1>
  <p>
    Hola, soy un subtitulo
  </p>
</Fragment>
`;
